datastream-server-restli/src/main/java/com/linkedin/datastream/server/dms/DatastreamResources.java

```diff
@@ -686,6 +686,37 @@ public Datastream get(String name) {
     return stream;
   }
 
+  /**
+   * Finds the task's assignment node which hosts the given task 
+   * @param pathKeys Datastream resource key
+   * @param task Datastream task name
+   */
+  @Action(name = "getTaskAssignment")
+  public String getTaskAssignment(@PathKeysParam PathKeys pathKeys, @ActionParam("task") String task) {
+    // Get datastream
+    String datastreamName = pathKeys.getAsString(KEY_NAME);
+
+    Datastream stream;
+    String assignedInstance = null;
+    try {
+      LOG.info("Get assigned instance for task: {}", task);
+      _dynamicMetricsManager.createOrUpdateMeter(CLASS_NAME, GET_CALL, 1);
+      stream = _store.getDatastream(datastreamName);
```

Comment:
Consider if you can encapsulate this logic inside store itself, rather than exposing the rawData outside.

Reply:
Instead of making multiple customized functions to retrieve specific data from the store, I wanted to make a generic function to the store which the brooklin-server can customize and handle.